run-tests.py

```diff
@@ -316,6 +323,50 @@ def get_all_documented_symbols():
     return documented_symbols
 
 
+def doctest_slow_books():
+    """
+    Check that all notebooks listed in .slow-books actually exist. This
+    prevents cron jobs falling over if slow notebooks are renamed without
+    editing the list.
+    """
+
+    print('\nChecking that all notebooks listed in .slow-books exist.')
+
+    with open('.slow-books', 'r') as f:
+        slow_books = [l.strip() for l in f.readlines() if
+                      l.strip().endswith('.ipynb')]
+
+    if len(slow_books) < 1:
+        print('No slow books found in .slow-books. Did something change?')
+        print('FAILED')
+        sys.exit(1)
+
+    with open('.slow-books', 'r') as f:
+        other_lines = [l.strip() for l in f.readlines() if not (
+            l.strip().startswith('#') or
+            l.strip().endswith('.ipynb') or
+            l.strip() == ''
+        )]
+
+    if len(other_lines) > 0:
+        print('The following entries are in .slow-books but are not ipynb:')
+        print('  {}'.format('\n  '.join(other_lines)))
+        print('FAILED')
+        sys.exit(1)
+
+    examples = [l for l in os.listdir('examples') if l.endswith('ipynb')]
+    undocumented = [b for b in slow_books if b not in examples]
```

Comment:
Maybe use sets here?

Reply:
Maybe next time ;).  I personally find the `difference()` method on sets difficult to read because I always have to check whether it's `A-B` or `B-A`.  I guess here it would be clear from the context, but I've tended to avoid it in the past.